src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -630,8 +630,8 @@ private static int lengthOfSoftClipping(Iterator<CigarElement> iterator) {
         int clipped = 0;
         while (iterator.hasNext()) {
             final CigarElement elem = iterator.next();
-            if (elem.getOperator() != CigarOperator.SOFT_CLIP) break;
-            clipped = elem.getLength();
+            if (elem.getOperator() != CigarOperator.SOFT_CLIP && elem.getOperator() != CigarOperator.HARD_CLIP) break;
+            if (elem.getOperator() == CigarOperator.SOFT_CLIP) clipped = elem.getLength();
```

Comment:
I think this is fine that we will lose the hard clip, since methods in `CigarUtil` also note this.  Perhaps add a TODO here and where this function is used that it doesn't handle hard clips?